test/core/consensus/validation/block_validator_test.cpp

```diff
@@ -0,0 +1,136 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "consensus/validation/babe_block_validator.hpp"
+
+#include <gtest/gtest.h>
+#include "mock/core/blockchain/block_tree_mock.hpp"
+#include "mock/core/runtime/tagged_transaction_queue_mock.hpp"
+#include "mock/crypto/hasher.hpp"
+#include "scale/scale.hpp"
+#include "testutil/literals.hpp"
+
+using namespace kagome;
+using namespace blockchain;
+using namespace consensus;
+using namespace runtime;
+using namespace hash;
+using namespace primitives;
+using namespace common;
+using namespace crypto;
+
+using testing::Return;
+
+class BlockValidatorTest : public testing::Test {
+ public:
+  std::pair<Seal, Blob<SR25519_PUBLIC_SIZE>> sealBlock(
+      Block &block, gsl::span<const uint8_t> block_hash) const {
+    // generate a new keypair
+    std::array<uint8_t, SR25519_SIGNATURE_SIZE> sr25519_signature{};
+
+    std::array<uint8_t, SR25519_KEYPAIR_SIZE> sr25519_keypair{};
+    std::array<uint8_t, SR25519_SEED_SIZE> seed{};
+    sr25519_keypair_from_seed(sr25519_keypair.data(), seed.data());
+
+    gsl::span<uint8_t, SR25519_KEYPAIR_SIZE> keypair_span(sr25519_keypair);
+    auto secret = keypair_span.subspan(0, SR25519_SECRET_SIZE);
+    auto pub_key =
+        keypair_span.subspan(SR25519_SECRET_SIZE, SR25519_PUBLIC_SIZE);
+
+    // create a signature
+    sr25519_sign(sr25519_signature.data(),
+                 pub_key.data(),
+                 secret.data(),
+                 block_hash.data(),
+                 block_hash.size());
+
+    // seal the block
+    Seal seal{sr25519_signature};
+    block.header.digests.emplace_back(scale::encode(seal).value());
+
+    Blob<SR25519_PUBLIC_SIZE> pubkey_blob{};
+    std::copy(pub_key.begin(), pub_key.end(), pubkey_blob.begin());
+    return {seal, pubkey_blob};
+  }
+
+  // fields for validator
+  std::shared_ptr<BlockTreeMock> tree_ = std::make_shared<BlockTreeMock>();
+  std::shared_ptr<TaggedTransactionQueueMock> tx_queue_ =
+      std::make_shared<TaggedTransactionQueueMock>();
+  std::shared_ptr<HasherMock> hasher_ = std::make_shared<HasherMock>();
+
+  BabeBlockValidator validator_{tree_, tx_queue_, hasher_};
+
+  // fields for block
+  Hash256 parent_hash_ =
+      Hash256::fromString("c30ojfn4983u4093jv3894j3f034oji").value();
+
+  BabeSlotNumber slot_number_ = 2;
+  VRFValue vrf_value_ = 1488228;
+  VRFProof vrf_proof_{
```

Comment:
actually it can be any array of size 32. You can replace with
```
VRFProof vrf_proof_{};
vrf_proof_.fill(0);
```
to make it less verbose

Reply:
although with such approach you would need to put initialization into setUp method